File: test_commits/python/refactor_changed4.py
def log_login_time(user):
    print(f"{user.name} logged in at {user.login_time}")

def log_session_duration(user):
    session_length = (user.logout_time - user.login_time).seconds
    print(f"Session duration: {session_length} seconds")

def warn_failed_attempts(user):
    if user.failed_attempts > 3:
        print("User had multiple failed login attempts")

def warn_password_expiry(user):
    if (user.login_time - user.password_last_changed).days > 90:
        print("User password may be expired")

def report_user_activity(user):
    log_login_time(user)
    log_session_duration(user)
    warn_failed_attempts(user)
    warn_password_expiry(user)

File: test_commits/javascript/refactor_changed4.js
function logLoginTime(user) {
    console.log(`${user.name} logged in at ${user.loginTime}`);
}

function logSessionDuration(user) {
    const sessionLength = (user.logoutTime - user.loginTime) / 1000;
    console.log(`Session duration: ${sessionLength} seconds`);
}

function warnFailedAttempts(user) {
    if (user.failedAttempts > 3) {
        console.log("User had multiple failed login attempts");
    }
}

function warnPasswordExpiry(user) {
    const passwordAge = (user.loginTime - user.passwordLastChanged) / (1000 * 60 * 60 * 24);
    if (passwordAge > 90) {
        console.log("User password may be expired");
    }
}

function reportUserActivity(user) {
    logLoginTime(user);
    logSessionDuration(user);
    warnFailedAttempts(user);
    warnPasswordExpiry(user);
}
